File: examples/thymeleaf-java-example/src/main/java/de/tschuehly/example/thymeleafjava/web/index/IndexViewComponent.java
package de.tschuehly.example.thymeleafjava.web.index;

import de.tschuehly.spring.viewcomponent.core.component.ViewComponent;
import de.tschuehly.spring.viewcomponent.thymeleaf.ViewContext;

@ViewComponent
public class IndexViewComponent {
    public ViewContext render() {
        return new IndexView();
    }

    public record IndexView() implements ViewContext {
    }
}

File: examples/thymeleaf-kotlin-example/src/main/kotlin/de/tschuehly/example/thymeleafkotlin/web/index/IndexViewComponent.kt
package de.tschuehly.example.thymeleafkotlin.web.index

import de.tschuehly.spring.viewcomponent.core.component.ViewComponent
import de.tschuehly.spring.viewcomponent.thymeleaf.ViewContext

@ViewComponent
class IndexViewComponent {
    fun render() = IndexView()

    class IndexView : ViewContext
}